core/blockchain.go

```diff
@@ -1480,13 +1462,18 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		// Falls through to the block import
 	}
 	switch {
+	// We're post-merge and the parent is pruned, try to recover the parent state
+	case !setHead && errors.Is(err, consensus.ErrPrunedAncestor):
+		log.Debug("Pruned ancestor", "number", block.Number(), "hash", block.Hash())
+		return it.index, bc.recoverAncestors(block)
+
 	// First block is pruned, insert as sidechain and reorg only if TD grows enough
-	case errors.Is(err, consensus.ErrPrunedAncestor):
+	case setHead && errors.Is(err, consensus.ErrPrunedAncestor):
 		log.Debug("Pruned ancestor, inserting as sidechain", "number", block.Number(), "hash", block.Hash())
 		return bc.insertSideChain(block, it)
 
 	// First block is future, shove it (and all children) to the future queue (unknown ancestor)
-	case errors.Is(err, consensus.ErrFutureBlock) || (errors.Is(err, consensus.ErrUnknownAncestor) && bc.futureBlocks.Contains(it.first().ParentHash())):
+	case setHead && (errors.Is(err, consensus.ErrFutureBlock) || (errors.Is(err, consensus.ErrUnknownAncestor) && bc.futureBlocks.Contains(it.first().ParentHash()))):
```

Comment:
Is this change needed? 
This deals with future blocks. Which can happen in two cases: 
- For the current one we're trying to import, the consensus engine gave us `ErrFutureBlock`. This should not happen with the new merger-engine, right?
- If the current one we're trying to import is missing a parent, and that parent is found within the cache of futureBlocks. I guess this _can_ happen, if e.g. someone mines the terminal block with a future timestamp, and then someone mines a PoS-block on top. But in that case, isn't it correct (ish) to enter here? Either way, I'm not convinced that it's correct to just ignore the clause